.github/workflows/builds.yml

```diff
@@ -107,7 +107,32 @@ jobs:
           password: ${{ secrets.DOCKERHUB_TOKEN }}
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
+      - name: Build image (no-push)
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          platforms: ${{ matrix.build-config.platform }}
+          push: false
+          tags: ${{ matrix.build-config.tags }}
+          labels: ${{ matrix.build-config.labels }}
+          build-args: |
+            BUILD_DATE=${{ needs.docker-oss-meta.outputs.build-date }}
+            VERSION=${{ needs.docker-oss-meta.outputs.version }}
+            COMMIT_HASH=${{ github.sha }}
+            REVISION=${{ needs.docker-oss-meta.outputs.revision }}
+          outputs: type=docker,dest=/tmp/local-rudder-server.tar
+          cache-to: type=local,dest=/tmp/.buildx-cache,mode=max,tag=latest
```

Comment:
should we use some other tag here instead of latest?

Reply:
We could do that, but it doesn't really matter as long as cache-to and cache-from tags matching. It only affects how things are stored/retrieved from this temporary cache

Reply:
Does it make sense if I rename it to `temp` ?

Reply:
The only doubt i have is if 2 workflows run on same machine, then is there a possibility tags may get overriden by other workflows?

Reply:
Good point. I opted for `${{ runner.temp }}` instead of `/tmp` because it offers more reliable behavior in GitHub Actions workflows. Here's why:

1. Guaranteed uniqueness: `${{ runner.temp }}` provides a unique directory for each job, preventing potential conflicts.
2. Automatic cleanup: GitHub ensures this directory is cleaned after each job, reducing leftover artifacts.
3. Consistent across platforms: It works uniformly across different runner operating systems.